### PR TITLE
Internal: clean Translation_unit

### PR DESCRIPTION
**Pull request by gpetiot**
No functional change, only refactoring/cleaning/trying to improve the code quality.
Each commit can be reviewed independently.

### CODE REVIEW

File: src/Translation_unit.ml [hunk 1]
```diff
@@ -91,7 +91,8 @@ let dump_formatted ~input_name ?output_file ~suffix fmted =
   else None
 
 let print_error ?(quiet_unstable = false) ?(quiet_comments = false)
-    ?(quiet_doc_comments = false) conf fmt input_name error =
+    ?(quiet_doc_comments = false) ?(fmt = Format.err_formatter) conf
+    ~input_name error =
```
Comment (Julow) [inline src/Translation_unit.ml]: `quiet_unstable`, `quiet_comments` and `quiet_doc_comments` are never used. They can be removed and `quiet_exn` below can be simplified to `false`.

File: src/Translation_unit.ml [hunk 2]
```diff
@@ -365,23 +361,16 @@ let format xunit (conf : Conf.t) ?output_file ~input_name ~source ~parsed ()
...
-      | Ok t -> (
-        try print_check ~i:1 ~conf t ~source with
-        | Sys_error msg -> Error (User_error msg)
-        | exn -> Error (Ocamlformat_bug {exn}) )
-  in
-  ( match result with
-  | Ok _ -> ()
-  | Error e -> print_error conf Format.err_formatter input_name e ) ;
-  result
+  if conf.disable then Ok source
```
Comment (Julow) [inline src/Translation_unit.ml]: This function does not seem to do anything when `conf.disable` is `true`. Same when `parsed` is `Error _`. What do you think of moving these checks into `parse_and_format` below ?
Also, we can avoid parsing when `conf.disable` is `true`.

File: src/Translation_unit.ml (hunk 1 above)
Comment (gpetiot) [reply src/Translation_unit.ml]: done

File: src/Translation_unit.ml (hunk 2 above)
Comment (gpetiot) [reply src/Translation_unit.ml]: indeed we can, done.

File: src/Translation_unit.ml [hunk 3]
```diff
@@ -365,23 +346,14 @@ let format xunit (conf : Conf.t) ?output_file ~input_name ~source ~parsed ()
...
-        try print_check ~i:1 ~conf t ~source with
-        | Sys_error msg -> Error (User_error msg)
-        | exn -> Error (Ocamlformat_bug {exn}) )
-  in
-  ( match result with
-  | Ok _ -> ()
-  | Error e -> print_error conf Format.err_formatter input_name e ) ;
-  result
+  match parsed with
+  | Error exn -> Error (Invalid_source {exn})
```
Comment (Julow) [inline src/Translation_unit.ml]: This should be checked in `parse_and_format` below.
Does not make sense to have a `result` here.

Comment (gpetiot) [reply src/Translation_unit.ml]: indeed, done

**Pull request merged**